internal/version/version.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+/*
+Package version provides functionality to parse and process semantic versions, as they are used in multiple components of Constellation.
+*/
+package version
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"golang.org/x/mod/semver"
+)
+
+// Version represents a semantic version.
+type Version struct {
```

Comment:
Rename this to `Semver`? Package could also be renamed to `semver`. Would remove the naming conflict with the type from `internal/versionsapi` and would partially address @Nirusu comment.